### ISSUE TITLE
Breaking change introduced in 2.10.0

### ISSUE THREAD

**Issue by hudochenkov** [molefrog/wouter#292]
Version 2.10.0 [uses](https://github.com/molefrog/wouter/blob/efda2abc95bdb8b01d9d55b865078fbae63c4a2b/react-deps.js#L5) `useInsertionEffect`, which is not available in React 17. It breaks the build:

> ERROR in ./node_modules/wouter/react-deps.js 46:2-27
>
> export 'useInsertionEffect' (imported as 'useBuiltinInsertionEffect') was not found in 'react' (possible exports: Children, Component, Fragment, Profiler, PureComponent, StrictMode, Suspense, __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED, cloneElement, createContext, createElement, createFactory, createRef, forwardRef, isValidElement, lazy, memo, useCallback, useContext, useDebugValue, useEffect, useImperativeHandle, useLayoutEffect, useMemo, useReducer, useRef, useState, version)

In the error webpack can't find export, which is correct, since React 17 [doesn't have](https://17.reactjs.org/docs/hooks-reference.html) this hook.

**Comment by cbbfcd** [comment]
You can go back to v2.9.2 first. maybe should add some description about it in `readme`

**Comment by HansBrende** [comment]
I _believe_ this can be fixed simply by changing

```javascript
import {
  useEffect,
  useLayoutEffect,
  useRef,
  useInsertionEffect as useBuiltinInsertionEffect,
} from "react";
```

to:

```javascript
import * as React from "react";
const {useEffect, useLayoutEffect, useRef, useInsertionEffect: useBuiltinInsertionEffect} = React;
```

The intention was for useInsertionEffect to be `undefined` on React 17, not throw an error. (And if so, use `useLayoutEffect` instead).

**Comment by molefrog** [comment]
Thank you for reporting this. Has been fixed in v2.10.1
https://github.com/molefrog/wouter/commit/f9333dc7bb9c3eab1f3025bcb0ca03b49cb837a5
